### ISSUE TITLE
standardize validator functions to all `Object` props 

### ISSUE THREAD

**Issue by indirectlylit** [learningequality/kolibri#8903]
## Observed behavior

Following up from: https://github.com/learningequality/kolibri/issues/8640, https://github.com/learningequality/kolibri/pull/8878, and https://github.com/learningequality/kolibri/pull/8902 which added validation functions.

There are a large number of `Object`-type props with no validator function. (Search the codebase using a multi-line regex to help find them: `{[^}]*type: Object[^}]*}`)

Normal component props are well-documented using `type`, `required`, and sometimes `validator`. The keys of `Object` props are as much a part of the component API as the other ones, but they are often under-documented.

Even when validation of `Object` props is done in Kolibri, it is performed in an ad-hoc manner with little consistency and the logging is less useful because usually Vue will simply say the validator failed without specifying why. Some examples:

https://github.com/learningequality/kolibri/blob/c3238a85915a75386150c5f1191de6735e495d1c/kolibri/core/assets/src/views/sync/FacilityAdminCredentialsForm.vue#L54-L60

https://github.com/learningequality/kolibri/blob/c3238a85915a75386150c5f1191de6735e495d1c/kolibri/plugins/setup_wizard/assets/src/views/importFacility/SelectFacilityForm.vue#L67-L73

https://github.com/learningequality/kolibri/blob/c3238a85915a75386150c5f1191de6735e495d1c/kolibri/plugins/learn/assets/src/views/EmbeddedSidePanel/SelectGroup.vue#L72-L79

## Expected behavior

Component object props should have a validator and fill in default values which should simplify internal logic of components using these props

**Comment by nikkuAg** [comment]
Is this issue still open? I would like to work on this

**Comment by MisRob** [comment]
Hi, @nikkuAg, yes, thanks for volunteering

**Comment by MisRob** [comment]
This seems to be relevant https://github.com/learningequality/kolibri/pull/8902/files#diff-294e45dbea9c24d93f492cafd5cf86476c419f189cbb3bf0d77f7ffb9f8ac220

**Comment by indirectlylit** [comment]
> This seems to be relevant

yes, that was the intent - thank you

> Introduces two new functions, `validateObject` and `objectWithDefaults`, which bring Vue prop-like validation and default behavior to nested javascript objects. This can be used with component props, vuex and API validation, and other scenarios

a stopgap for lack of TypeScript, perhaps :)

**Comment by rtibbles** [comment]
> a stopgap for lack of TypeScript

Except we could use this for runtime validation, which TypeScript does not provide!

**Comment by nikkuAg** [comment]
Hi,
I am not entirely able to understand what I have to do.

**Comment by rtibbles** [comment]
The basic idea is to use the Object validation specification work that was done in the referenced PRs, and use it to fully document each of the cases where an Object type is used as a prop.

This allows for better specification of the intended properties and shape of the prop, and to give more semantic error messages when these expectations are violated.

I would start by looking at one of the examples listed in the issue, and replacing the validator function with the Object validation specification.

**Comment by MisRob** [comment]
Hi @nikkuAg, are you still planning to work on this at some point or would it be better to unassign?

**Comment by nikkuAg** [comment]
Hi @MisRob,
I'm currently occupied with another task, but once I complete it, I'll be able to pick this up. Therefore, I believe it would be best to unassign me for now so that someone else can take over.

**Comment by MisRob** [comment]
Okay @nikkuAg, thank you, let us know any time if you'd like to return to this

**Comment by tanouh** [comment]
Hello, if no one is assigned and it is not solved yet, I'd be pleased to take a  look on it !

**Comment by tiffcoding** [comment]
hello, can i be assigned to work on a few of these validators?

**Comment by rtibbles** [comment]
Hi @tiffcoding - yes, and please do feel free to open a PR with just a few of them, there is no need to do all of these all at once!

**Comment by MisRob** [comment]
Hi @tiffcoding, I wanted to mention that [Learning Equality will be closed from December 23 to January 5](https://github.com/learningequality/kolibri/discussions/12956).

**Comment by Ammog-Warrier** [comment]
Hello there. Is this still being worked on?Asking since its been 3 weeks. I'm open to working on it if it's open. Thank You

**Comment by akolson** [comment]
HI @Ammog-Warrier!

Thank you for your interest in working on this issue. However, @tiffcoding has already opened a pr for the same [here](https://github.com/learningequality/kolibri/pull/12921). Please be sure to look out for issues marked with the label "help wanted" that are suitable for contribution. You can find them [here](https://github.com/search?q=org%3Alearningequality+is%3Aopen+is%3Aissue+label%3A%22help+wanted%22+no%3Aassignee&type=issues).

Thanks again and we are looking forward to your contribution.
_(Just a small note that we might be slow to respond as we are closed for the year, but should be able to respond to any of your queries once we are back fully, next year)_

**Comment by tiffcoding** [comment]
Hello, I apologize for the delay! Due to holidays, i have not been able to work on this. feel free to hand it off to Ammog-Warrior for progress!

**Comment by akolson** [comment]
Thank you @tiffcoding! 
@Ammog-Warrier, Please confirm if you are in position to work on this issue so I can assign it to you, thank you!

**Comment by Ammog-Warrier** [comment]
Yea I'd like to work on it if possible. Thank You

**Comment by akolson** [comment]
Great! Thank you! I'll assign it to you shortly!

**Comment by nucleogenesis** [comment]
@Ammog-Warrier do you still plan to tackle this issue?

In any case, @tiffcoding 's work in #12921 may be helpful for whoever ends up working on this.

**Comment by Ammog-Warrier** [comment]
> @Ammog-Warrier do you still plan to tackle this issue?
> 
> In any case, @tiffcoding 's work in #12921 may be helpful for whoever ends up working on this.

Yes yes, I'm currently working on it. I just glt sidetracked for few days because of some life stuff. Thank you for the advice. I'll look into it

**Comment by akolson** [comment]
Thank you @Ammog-Warrier for your confirmation!

**Comment by Ammog-Warrier** [comment]
Hey there! sorry for this, but I was free for a bit but my schedule just got even worse loaded than I expected. I sincerely apologise and request you to assign this issue to someone else.

**Comment by Abhishek-Punhani** [comment]
@MisRob  Can i work on this issue?

**Comment by MisRob** [comment]
Hi @Ammog-Warrier, no problem  at all, thank you for letting us know.

Hi @Abhishek-Punhani, yes I will assign you. Thanks!

**Comment by MisRob** [comment]
@Abhishek-Punhani I'd recommend you study https://github.com/learningequality/kolibri/pull/12921 and @nucleogenesis's review - it has helpful information for this task

**Comment by Abhishek-Punhani** [comment]
@MisRob  I was looking into this issue and I found that changes could be done ContentPage.vue,SelectGroup.vue and FacilityAdminCredentialsForm.vue
I didnt find any validator used in kolibri/plugins/setup_wizard/assets/src/views/importFacility/SelectFacilityForm.vue

![Image](https://github.com/user-attachments/assets/5a058180-8b1e-4b13-9612-e20cd53a55f0)

Is this what I am supposed to do?

**Comment by MisRob** [comment]
Hi @Abhishek-Punhani, I think there's many many places to update in Kolibri, basically any time there's an `Object` `prop`. I'd definitely recommend your start with just a very small number of files for your first pull request. After we confirm it's all good, you can continue with more places if you wish to. Your screenshot looks like a good direction to me.

**Comment by MisRob** [comment]
@Abhishek-Punhani We don't only want to update existing validators but also to validate objects that don't have any validators yet. But as said, in smaller chunks ideally :) Thanks!

**Comment by Abhishek-Punhani** [comment]
@MisRob Thank you for your feedback. I’ve raised a PR where I modified the validators function in three Vue files to incorporate the validateObject function. Please review it at your convenience. I believe I can implement the validateObject and objectWithDefaults functions in upcoming PRs, but I would greatly appreciate your guidance to ensure they are implemented effectively.

**Comment by MisRob** [comment]
Hi @Abhishek-Punhani, thank you! I think you can feel free to continue working on this issue.

**Comment by Abhishek-Punhani** [comment]
@MisRob  Thanks! Since this approach has been applied to a few files in the merged PR, should I continue making similar changes across other relevant files, or would you prefer a different approach? Let me know how you'd like me to proceed!

**Comment by MisRob** [comment]
Yes @Abhishek-Punhani, feel free to gradually work through this as you wish. The only thing I'd recommend opening pull requests of reasonable size since reviews will be easier and faster than having a huge PR, and that's all. However you can definitely do a bit more than in the first one, perhaps around 10 files in a single PR? That's just my opinionated recommendation, you're welcome to choose.

**Comment by Abhishek-Punhani** [comment]
Got it . Thanks @MisRob . I will try to raise a PR soon

**Comment by Abhishek-Punhani** [comment]
Hi @MisRob  I think we can begin by resolving all the files in the Learn App. I have raised a PR modifying 10 files—please review it at your convenience.

**Comment by MisRob** [comment]
Sure, thanks @Abhishek-Punhani

**Comment by EshaanAgg** [comment]
Hi! Since the scope of this issue is the entire codebase, would it be okay if I, too, joined to help? I can coordinate with @Abhishek-Punhani to work on disjoint directories, so that there is no duplication of efforts and we can work on the issue parallely.

**Comment by Abhishek-Punhani** [comment]
Hi @EshaanAgg , I think it would it better if we both collaborate together and coordinate to avoid any conflicts in our PRs. Let me know how you'd like to split things up! I think with this we can close this issue at a faster pace . @MisRob Please let me know how would you  like me and @EshaanAgg  to proceed with this issue!

**Comment by EshaanAgg** [comment]
I was able to jot up a quick script that listed all the files that match the given regex and probably require intervention. One way to open up the issue to multiple contributors would be to treat this message (or we can copy this into the issue description itself) and treat it as the issue tracker. Any contributor can message here to claim a folder that he/she wants to work on, and the maintainers can edit the message to add a `WIP @Contributor` in front of the folder to indicate the same. Once the folder is done, we can delete it from the markdown or check the checkbox! 

How does this approach sound?

### Files with Object-Type Props Missing Validators

- [ ] [coach](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach)
  - [ ] [assets](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets)
    - [ ] [src](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src)
      - [ ] [views](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views)
        - [ ] [CoachImmersivePage.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/CoachImmersivePage.vue)
        - [ ] [common](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/common)
          - [ ] [BackLink.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/common/BackLink.vue)
          - [ ] [HeaderTabs](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/common/HeaderTabs)
            - [ ] [HeaderTab.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/common/HeaderTabs/HeaderTab.vue)
          - [ ] [LessonStatus.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/common/LessonStatus.vue)
          - [ ] [QuizLessonDetailsHeader.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/common/QuizLessonDetailsHeader.vue)
          - [ ] [QuizStatus.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/common/QuizStatus.vue)
          - [ ] [ReportsResourceHeader.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/common/ReportsResourceHeader.vue)
          - [ ] [assignments](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/common/assignments)
            - [ ] [AssignmentDetailsModal.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/common/assignments/AssignmentDetailsModal.vue)
          - [ ] [notifications](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/common/notifications)
            - [ ] [NotificationCard.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/common/notifications/NotificationCard.vue)
          - [ ] [resourceSelection](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/common/resourceSelection)
            - [ ] [QuizResourceSelectionHeader.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/common/resourceSelection/QuizResourceSelectionHeader.vue)
            - [ ] [UpdatedResourceSelection.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/common/resourceSelection/UpdatedResourceSelection.vue)
            - [ ] [subPages](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/common/resourceSelection/subPages)
              - [ ] [PreviewSelectedResources](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/common/resourceSelection/subPages/PreviewSelectedResources)
                - [ ] [PreviewContent.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/common/resourceSelection/subPages/PreviewSelectedResources/PreviewContent.vue)
                - [ ] [index.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/common/resourceSelection/subPages/PreviewSelectedResources/index.vue)
              - [ ] [SelectFromTopicTree.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/common/resourceSelection/subPages/SelectFromTopicTree.vue)
        - [ ] [groups](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/groups)
          - [ ] [GroupsRootPage](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/groups/GroupsRootPage)
            - [ ] [GroupRow.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/groups/GroupsRootPage/GroupRow.vue)
        - [ ] [home](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/home)
          - [ ] [HomePage](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/home/HomePage)
            - [ ] [Block.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/home/HomePage/Block.vue)
            - [ ] [ItemProgressDisplay.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/home/HomePage/ItemProgressDisplay.vue)
        - [ ] [lessons](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/lessons)
          - [ ] [LessonResourceSelectionPage](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/lessons/LessonResourceSelectionPage)
            - [ ] [LessonContentCard](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/lessons/LessonResourceSelectionPage/LessonContentCard)
              - [ ] [index.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/lessons/LessonResourceSelectionPage/LessonContentCard/index.vue)
            - [ ] [SearchTools](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/lessons/LessonResourceSelectionPage/SearchTools)
              - [ ] [LessonsSearchFilters.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/lessons/LessonResourceSelectionPage/SearchTools/LessonsSearchFilters.vue)
          - [ ] [LessonSelectionContentPreviewPage](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/lessons/LessonSelectionContentPreviewPage)
            - [ ] [LessonContentPreview](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/lessons/LessonSelectionContentPreviewPage/LessonContentPreview)
              - [ ] [ContentArea.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/lessons/LessonSelectionContentPreviewPage/LessonContentPreview/ContentArea.vue)
              - [ ] [index.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/lessons/LessonSelectionContentPreviewPage/LessonContentPreview/index.vue)
          - [ ] [LessonSummaryPage](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/lessons/LessonSummaryPage)
            - [ ] [sidePanels](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/lessons/LessonSummaryPage/sidePanels)
              - [ ] [LessonResourceSelection](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/lessons/LessonSummaryPage/sidePanels/LessonResourceSelection)
                - [ ] [subPages](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/lessons/LessonSummaryPage/sidePanels/LessonResourceSelection/subPages)
                  - [ ] [SearchFilters.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/lessons/LessonSummaryPage/sidePanels/LessonResourceSelection/subPages/SearchFilters.vue)
                  - [ ] [SelectFromSearchResults.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/lessons/LessonSummaryPage/sidePanels/LessonResourceSelection/subPages/SelectFromSearchResults.vue)
        - [ ] [quizzes](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/quizzes)
          - [ ] [CreateExamPage](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/quizzes/CreateExamPage)
            - [ ] [QuestionListPreview.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/quizzes/CreateExamPage/QuestionListPreview.vue)
            - [ ] [sidePanels](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/quizzes/CreateExamPage/sidePanels)
              - [ ] [QuizResourceSelection](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/quizzes/CreateExamPage/sidePanels/QuizResourceSelection)
                - [ ] [subPages](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/quizzes/CreateExamPage/sidePanels/QuizResourceSelection/subPages)
                  - [ ] [QuestionsSettings.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/quizzes/CreateExamPage/sidePanels/QuizResourceSelection/subPages/QuestionsSettings.vue)
          - [ ] [QuizSummaryPage](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/quizzes/QuizSummaryPage)
            - [ ] [ManageExamModals.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/coach/assets/src/views/quizzes/QuizSummaryPage/ManageExamModals.vue)
- [ ] [device](https://github.com/learningequality/kolibri/tree/develop/kolibri/device)
  - [ ] [assets](https://github.com/learningequality/kolibri/tree/develop/kolibri/device/assets)
    - [ ] [src](https://github.com/learningequality/kolibri/tree/develop/kolibri/device/assets/src)
      - [ ] [views](https://github.com/learningequality/kolibri/tree/develop/kolibri/device/assets/src/views)
        - [ ] [DeviceSettingsPage](https://github.com/learningequality/kolibri/tree/develop/kolibri/device/assets/src/views/DeviceSettingsPage)
          - [ ] [ServerRestartModal.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/device/assets/src/views/DeviceSettingsPage/ServerRestartModal.vue)
        - [ ] [FacilitiesPage](https://github.com/learningequality/kolibri/tree/develop/kolibri/device/assets/src/views/FacilitiesPage)
          - [ ] [FacilityAdminCredentialsModal.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/device/assets/src/views/FacilitiesPage/FacilityAdminCredentialsModal.vue)
          - [ ] [RemoveFacilityModal.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/device/assets/src/views/FacilitiesPage/RemoveFacilityModal.vue)
        - [ ] [ManageContentPage](https://github.com/learningequality/kolibri/tree/develop/kolibri/device/assets/src/views/ManageContentPage)
          - [ ] [ChannelPanel](https://github.com/learningequality/kolibri/tree/develop/kolibri/device/assets/src/views/ManageContentPage/ChannelPanel)
            - [ ] [ChannelDetails.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/device/assets/src/views/ManageContentPage/ChannelPanel/ChannelDetails.vue)
            - [ ] [WithImportDetails.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/device/assets/src/views/ManageContentPage/ChannelPanel/WithImportDetails.vue)
          - [ ] [SelectionBottomBar.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/device/assets/src/views/ManageContentPage/SelectionBottomBar.vue)
          - [ ] [TasksBar.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/device/assets/src/views/ManageContentPage/TasksBar.vue)
        - [ ] [ManageTasksPage](https://github.com/learningequality/kolibri/tree/develop/kolibri/device/assets/src/views/ManageTasksPage)
          - [ ] [TaskPanel.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/device/assets/src/views/ManageTasksPage/TaskPanel.vue)
        - [ ] [SelectContentPage](https://github.com/learningequality/kolibri/tree/develop/kolibri/device/assets/src/views/SelectContentPage)
          - [ ] [ChannelContentsSummary.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/device/assets/src/views/SelectContentPage/ChannelContentsSummary.vue)
          - [ ] [ContentNodeRow.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/device/assets/src/views/SelectContentPage/ContentNodeRow.vue)
          - [ ] [ContentTreeViewer.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/device/assets/src/views/SelectContentPage/ContentTreeViewer.vue)
        - [ ] [WelcomeModal.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/device/assets/src/views/WelcomeModal.vue)
- [ ] [epub_viewer](https://github.com/learningequality/kolibri/tree/develop/kolibri/epub_viewer)
  - [ ] [assets](https://github.com/learningequality/kolibri/tree/develop/kolibri/epub_viewer/assets)
    - [ ] [src](https://github.com/learningequality/kolibri/tree/develop/kolibri/epub_viewer/assets/src)
      - [ ] [views](https://github.com/learningequality/kolibri/tree/develop/kolibri/epub_viewer/assets/src/views)
        - [ ] [SearchSideBar.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/epub_viewer/assets/src/views/SearchSideBar.vue)
        - [ ] [SettingsSideBar.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/epub_viewer/assets/src/views/SettingsSideBar.vue)
        - [ ] [TableOfContentsSection.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/epub_viewer/assets/src/views/TableOfContentsSection.vue)
- [ ] [facility](https://github.com/learningequality/kolibri/tree/develop/kolibri/facility)
  - [ ] [assets](https://github.com/learningequality/kolibri/tree/develop/kolibri/facility/assets)
    - [ ] [src](https://github.com/learningequality/kolibri/tree/develop/kolibri/facility/assets/src)
      - [ ] [views](https://github.com/learningequality/kolibri/tree/develop/kolibri/facility/assets/src/views)
        - [ ] [ManageClassPage](https://github.com/learningequality/kolibri/tree/develop/kolibri/facility/assets/src/views/ManageClassPage)
          - [ ] [ClassDeleteModal.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/facility/assets/src/views/ManageClassPage/ClassDeleteModal.vue)
- [ ] [learn](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn)
  - [ ] [assets](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets)
    - [ ] [src](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src)
      - [ ] [views](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views)
        - [ ] [AlsoInThis.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/AlsoInThis.vue)
        - [ ] [AssessmentWrapper](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/AssessmentWrapper)
          - [ ] [index.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/AssessmentWrapper/index.vue)
        - [ ] [BrowseResourceMetadata.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/BrowseResourceMetadata.vue)
        - [ ] [ChannelCard.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/ChannelCard.vue)
        - [ ] [ChannelRenderer](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/ChannelRenderer)
          - [ ] [ContentItem.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/ChannelRenderer/ContentItem.vue)
          - [ ] [CustomContentRenderer.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/ChannelRenderer/CustomContentRenderer.vue)
        - [ ] [CompletionModal](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/CompletionModal)
          - [ ] [CompletionModalSection.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/CompletionModal/CompletionModalSection.vue)
          - [ ] [ResourceItem](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/CompletionModal/ResourceItem)
            - [ ] [index.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/CompletionModal/ResourceItem/index.vue)
          - [ ] [index.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/CompletionModal/index.vue)
        - [ ] [ContentPage.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/ContentPage.vue)
        - [ ] [CurrentlyViewedResourceMetadata.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/CurrentlyViewedResourceMetadata.vue)
        - [ ] [ExamPage](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/ExamPage)
          - [ ] [AnswerHistory.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/ExamPage/AnswerHistory.vue)
        - [ ] [ExploreLibrariesPage](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/ExploreLibrariesPage)
          - [ ] [LibraryItem.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/ExploreLibrariesPage/LibraryItem.vue)
        - [ ] [HybridLearningContentCard](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/HybridLearningContentCard)
          - [ ] [CardThumbnail.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/HybridLearningContentCard/CardThumbnail.vue)
          - [ ] [HybridLearningFooter.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/HybridLearningContentCard/HybridLearningFooter.vue)
        - [ ] [HybridLearningLessonCard.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/HybridLearningLessonCard.vue)
        - [ ] [LearningActivityDuration](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/LearningActivityDuration)
          - [ ] [index.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/LearningActivityDuration/index.vue)
        - [ ] [LibraryPage](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/LibraryPage)
          - [ ] [UnPinnedDevices.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/LibraryPage/UnPinnedDevices.vue)
        - [ ] [ProgressBar.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/ProgressBar.vue)
        - [ ] [QuizRenderer](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/QuizRenderer)
          - [ ] [QuizReport.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/QuizRenderer/QuizReport.vue)
          - [ ] [index.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/QuizRenderer/index.vue)
        - [ ] [SearchResultsGrid.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/SearchResultsGrid.vue)
        - [ ] [TopicsPage](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/TopicsPage)
          - [ ] [ToggleHeaderTabs.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/TopicsPage/ToggleHeaderTabs.vue)
          - [ ] [TopicSubsection.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/TopicsPage/TopicSubsection.vue)
          - [ ] [TopicsMobileHeader.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/TopicsPage/TopicsMobileHeader.vue)
        - [ ] [cards](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/cards)
          - [ ] [BaseCard.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/cards/BaseCard.vue)
          - [ ] [CardLink.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/cards/CardLink.vue)
          - [ ] [LessonCard](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/cards/LessonCard)
            - [ ] [index.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/cards/LessonCard/index.vue)
          - [ ] [QuizCard](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/cards/QuizCard)
            - [ ] [index.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/cards/QuizCard/index.vue)
          - [ ] [ResourceCard.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/cards/ResourceCard.vue)
        - [ ] [thumbnails](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/thumbnails)
          - [ ] [ContentNodeThumbnail.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/learn/assets/src/views/thumbnails/ContentNodeThumbnail.vue)
- [ ] [media_player](https://github.com/learningequality/kolibri/tree/develop/kolibri/media_player)
  - [ ] [assets](https://github.com/learningequality/kolibri/tree/develop/kolibri/media_player/assets)
    - [ ] [src](https://github.com/learningequality/kolibri/tree/develop/kolibri/media_player/assets/src)
      - [ ] [views](https://github.com/learningequality/kolibri/tree/develop/kolibri/media_player/assets/src/views)
        - [ ] [MediaPlayerTranscript](https://github.com/learningequality/kolibri/tree/develop/kolibri/media_player/assets/src/views/MediaPlayerTranscript)
          - [ ] [TranscriptCue.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/media_player/assets/src/views/MediaPlayerTranscript/TranscriptCue.vue)
- [ ] [pdf_viewer](https://github.com/learningequality/kolibri/tree/develop/kolibri/pdf_viewer)
  - [ ] [assets](https://github.com/learningequality/kolibri/tree/develop/kolibri/pdf_viewer/assets)
    - [ ] [src](https://github.com/learningequality/kolibri/tree/develop/kolibri/pdf_viewer/assets/src)
      - [ ] [views](https://github.com/learningequality/kolibri/tree/develop/kolibri/pdf_viewer/assets/src/views)
        - [ ] [PdfPage.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/pdf_viewer/assets/src/views/PdfPage.vue)
        - [ ] [SideBar](https://github.com/learningequality/kolibri/tree/develop/kolibri/pdf_viewer/assets/src/views/SideBar)
          - [ ] [Bookmarks](https://github.com/learningequality/kolibri/tree/develop/kolibri/pdf_viewer/assets/src/views/SideBar/Bookmarks)
            - [ ] [BookmarkItem.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/pdf_viewer/assets/src/views/SideBar/Bookmarks/BookmarkItem.vue)
- [ ] [setup_wizard](https://github.com/learningequality/kolibri/tree/develop/kolibri/setup_wizard)
  - [ ] [assets](https://github.com/learningequality/kolibri/tree/develop/kolibri/setup_wizard/assets)
    - [ ] [src](https://github.com/learningequality/kolibri/tree/develop/kolibri/setup_wizard/assets/src)
      - [ ] [views](https://github.com/learningequality/kolibri/tree/develop/kolibri/setup_wizard/assets/src/views)
        - [ ] [OnboardingStepBase.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/setup_wizard/assets/src/views/OnboardingStepBase.vue)
        - [ ] [onboarding-forms](https://github.com/learningequality/kolibri/tree/develop/kolibri/setup_wizard/assets/src/views/onboarding-forms)
          - [ ] [UserCredentialsForm.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/setup_wizard/assets/src/views/onboarding-forms/UserCredentialsForm.vue)
- [ ] [user_auth](https://github.com/learningequality/kolibri/tree/develop/kolibri/user_auth)
  - [ ] [assets](https://github.com/learningequality/kolibri/tree/develop/kolibri/user_auth/assets)
    - [ ] [src](https://github.com/learningequality/kolibri/tree/develop/kolibri/user_auth/assets/src)
      - [ ] [views](https://github.com/learningequality/kolibri/tree/develop/kolibri/user_auth/assets/src/views)
        - [ ] [FacilitySelect.vue](https://github.com/learningequality/kolibri/tree/develop/kolibri/user_auth/assets/src/views/FacilitySelect.vue)

**Comment by MisRob** [comment]
Hi @EshaanAgg, thanks a lot for helping to coordinate, this would be very helpful.

**Comment by Abhishek-Punhani** [comment]
Hi @MisRob , Since #13232 is merged, we can mark those as done in the tracker message. Also, I have raised a PR, for three files.
Please review it at your convenience. Thanks!

**Comment by MisRob** [comment]
Thanks @Abhishek-Punhani
